de-lib/src/main/java/org/iplantc/de/diskResource/client/views/search/cells/DiskResourceQueryForm.java

```diff
@@ -1,344 +1,65 @@
- * supplied to the constructor, the form will be initialized with given query template. If the default
- * constructor is used, a new template will be created.
- * <p/>
- * <p/>
- * When the user clicks the "Search" button;
- * <ol>
- * <li>The form will be validated
- * <ol>
- * <li>If the form is <b>invalid</b>, the validation errors will appear in the form and no other action
- * will occur.</li>
- * <li>Else, a {@link SubmitDiskResourceQueryEvent} will be fired with the form's current query template,
- * and this form will be hidden.</li>
- * </ol>
- * </li>
- * </ol>
- * <p/>
- * <p/>
- * When the user clicks the "" hyperlink;
- * <ol>
- * <li>The form will be validated
- * <ol>
- * <li>If the form is <b>invalid</b>, the validation errors will appear in the form and not other action
- * will occur.</li>
- * <li>Else, the user will be presented with a text field allowing them to set a name. Then, if the user
- * clicks "Save", a {@link org.iplantc.de.diskResource.client.events.search.SaveDiskResourceQueryClickedEvent}
- * will be fired with the form's current query template and this form will be hidden.</li>
- * </ol>
- * </li>
- * </ol>
- * FIXME Re-implement with UI-binder
- * @author jstroot
+ * A form the user can fill out to perform advanced searches in the Data window which utilize the search service
  */
-public class DiskResourceQueryForm extends Composite implements
-                                                     Editor<DiskResourceQueryTemplate>,
-                                                     SaveDiskResourceQueryClickedEvent.HasSaveDiskResourceQueryClickedEventHandlers,
-                                                     HasSubmitDiskResourceQueryEventHandlers,
-                                                     SaveDiskResourceQueryClickedEvent.SaveDiskResourceQueryClickedEventHandler,
-                                                     TagAddedEvent.TagAddedEventHandler,
-                                                     RemoveTagSelected.RemoveTagSelectedHandler {
-
-    public interface DiskResourceQueryFormAppearance {
-        SafeHtml getQueryTable();
-
-        int columnFormWidth();
-
-        int columnWidth();
-
-        String nameHas();
-
-        String emptyText();
-
-        String nameHasNot();
-
-        String createdWithin();
-
-        String modifiedWithin();
-
-        String metadataAttributeHas();
-
-        String metadataValueHas();
-
-        String ownedBy();
-
-        String emptyNameText();
-
-        String sharedWith();
-
-        String fileSizeGreater();
-
-        String fileSizeLessThan();
-
-        String sizeDropDownWidth();
-
-        String includeTrash();
-
-        String searchBtnText();
-
-        String emptyTimeText();
-
-        String fileNameClass();
-
-        String createdWithinClass();
+public class DiskResourceQueryForm extends Composite implements SearchView {
```

Comment:
Do you have to extend from Sencha's `Composite` ?. I used GWT's `IsWidget` instead and worked for me.

Reply:
I tried GWT's Composite, let me try IsWidget